ow_lander/scripts/action_trajectories.py

```diff
@@ -0,0 +1,1026 @@
+        goal_pose.orientation.y = goal_pose.orientation.y
+        goal_pose.orientation.z = goal_pose.orientation.z
+        goal_pose.orientation.w = goal_pose.orientation.w
+
+        # Ask the planner to generate a plan to the approximate joint values generated
+        # by kinematics builtin IK solver. For more insight on this issue refer to:
+        # https://github.com/nasa/ow_simulator/pull/60
+        if approximate:
+            move_arm.set_joint_value_target(goal_pose, True)
+        else:
+            move_arm.set_pose_target(goal_pose)
+
+        _, plan, _, _ = move_arm.plan()
+
+        if len(plan.joint_trajectory.points) == 0:  # If no plan found, abort
+            return False
+
+        return plan
+
+    def move_to_pre_trench_configuration_dig_circ(self, move_arm, robot, x_start, y_start):
+        """
+        :type move_arm: class 'moveit_commander.move_group.MoveGroupCommander'
+        :type robot: class 'moveit_commander.RobotCommander'
+        :type x_start: float
+        :type y_start: float
+        """
+        # Initilize to current position
+        joint_goal = move_arm.get_current_pose().pose
+        robot_state = robot.get_current_state()
+        move_arm.set_start_state(robot_state)
+
+        # Compute shoulder yaw angle to trench
+        alpha = math.atan2(y_start-constants.Y_SHOU, x_start-constants.X_SHOU)
+        h = math.sqrt(pow(y_start-constants.Y_SHOU, 2) +
+                      pow(x_start-constants.X_SHOU, 2))
+        l = constants.Y_SHOU - constants.HAND_Y_OFFSET
+        beta = math.asin(l/h)
+        # Move to pre trench position, align shoulder yaw
+        joint_goal = move_arm.get_current_joint_values()
+        joint_goal[constants.J_DIST_PITCH] = 0.0
+        joint_goal[constants.J_HAND_YAW] = 0.0
+        joint_goal[constants.J_PROX_PITCH] = -math.pi/2
+        joint_goal[constants.J_SHOU_PITCH] = math.pi/2
+        joint_goal[constants.J_SHOU_YAW] = alpha + beta
+
+        # If out of joint range, abort
+        if not is_shou_yaw_goal_in_range(joint_goal):
+            return False
+
+        joint_goal[constants.J_SCOOP_YAW] = 0
+
+        move_arm.set_joint_value_target(joint_goal)
+        _, plan, _, _ = move_arm.plan()
+        return plan
+
+    def dig_circular(self, move_arm, move_limbs, robot, moveit_fk, args, server_stop):
+        """
+        :type move_arm: class 'moveit_commander.move_group.MoveGroupCommander'
+        :type robot: class 'moveit_commander.RobotCommander'
+        :type moveit_fk: class moveit_msgs/GetPositionFK
+        :type args: List[bool, float, int, float, float, float]
+        """
+
+        circ_traj = None
+        circ_traj = RobotTrajectory()
+
+        x_start = args.x_start
+        y_start = args.y_start
+        depth = args.depth
+        parallel = args.parallel
+        ground_position = args.ground_position
+        server_stop.get_state()
+        if server_stop.stopped:
+                return False
```

Comment:
This is a little improvement over the previous code, but unless @AstroStucky 's refactor (OCEANWATER-904) is going to happen soon, I still think we should issue a ROS message when an action is stopped.  Instead of repeating that line many times, the simple refactor I proposed last time would be easy, e.g. a function like:
```
 def check_for_stop(self, action_name, server_stop):
        server_stop.get_state()
        if server_stop.stopped:
            rospy.loginfo('%s was stopped.', action_name);
            return True
        return False
````
and calling it instead, e.g.
``` 
if self.check_for_stop("dig_circular", server_stop):
            return False
```